bpf/go_nethttp.h

```diff
@@ -421,7 +425,27 @@ static __always_inline void roundTripStartHelper(struct pt_regs *ctx) {
         return;
     }
 
+    if (!url_ptr || !read_go_str("host",
+                                 url_ptr,
+                                 go_offset_of(ot, (go_offset){.v = _host_ptr_pos}),
+                                 &trace.host,
+                                 sizeof(trace.host))) {
+        bpf_dbg_printk("can't read http Request.URL.Host");
+        return;
+    }
+
+    if (!url_ptr || !read_go_str("scheme",
+                                 url_ptr,
+                                 go_offset_of(ot, (go_offset){.v = _scheme_ptr_pos}),
+                                 &trace.scheme,
+                                 sizeof(trace.scheme))) {
+        bpf_dbg_printk("can't read http Request.URL.Scheme");
+        return;
+    }
```

Comment:
```suggestion
    if (!url_ptr) {
        return;
     }
     
    if (!read_go_str("host",
                                 url_ptr,
                                 go_offset_of(ot, (go_offset){.v = _host_ptr_pos}),
                                 &trace.host,
                                 sizeof(trace.host))) {
        bpf_dbg_printk("can't read http Request.URL.Host");
        return;
    }

    if (!read_go_str("scheme",
                                 url_ptr,
                                 go_offset_of(ot, (go_offset){.v = _scheme_ptr_pos}),
                                 &trace.scheme,
                                 sizeof(trace.scheme))) {
        bpf_dbg_printk("can't read http Request.URL.Scheme");
        return;
    }
```

rationale: it go me a bit confused if we were checking for a side-effect on `url_ptr` after the first call

alternatively, check if `base_ptr` is not null in `read_go_str` & co.

Reply:
good idea, I'll refactor it.